mlflow/tracking/client.py

```diff
@@ -1709,6 +1727,54 @@ def log_table(
             # for data like {"inputs": "What is MLflow?"}
             except ValueError:
                 data = pd.DataFrame([data])
+
+        # Check if the column is a `PIL.Image.Image` or `mlflow.Image` object
+        # and save filepath
+        if len(data.select_dtypes(include=["object"]).columns) > 0:
+
+            def process_image(image):
+                # remove extension from artifact_file
+                table_name, _ = os.path.splitext(artifact_file)
+                # save image to path
+                filepath = posixpath.join("table_images", table_name, str(uuid.uuid4()))
+                image_filepath = filepath + ".png"
+                compressed_image_filepath = filepath + ".webp"
+                with self._log_artifact_helper(run_id, image_filepath) as artifact_path:
+                    image.save(artifact_path)
+
+                # save compressed image to path
+                compressed_image = compress_image_size(image)
+
+                with self._log_artifact_helper(run_id, compressed_image_filepath) as artifact_path:
+                    compressed_image.save(artifact_path)
+
+                # return a dictionary object indicating its an image path
+                return {
+                    "type": "image",
+                    "filepath": image_filepath,
+                    "compressed_filepath": compressed_image_filepath,
+                }
+
+            def check_is_image_object(obj):
+                return (
```

Comment:
can we do this? 

```
im.__class__.__module__ == "PIL.Image"
```

<img width="536" alt="image" src="https://github.com/mlflow/mlflow/assets/22925031/869a0208-f576-4678-a53c-9649126278ec">

Reply:
I think for this the problem is that sometimes PIL has other classes like `PIL.JpegImagePlugin`, so they don't always match to `PIL.Image`